synthesis/synthesis.go

```diff
@@ -260,76 +181,61 @@ func FixCds(sqlitePath string, sequence string, codontable codon.Table, problema
 				if codon.Triplet != toCodon.Triplet {
 					toCodonBias := strings.Count(toCodon.Triplet, "G") + strings.Count(toCodon.Triplet, "C")
 					switch {
-					case codonBias == toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "NA")
 					case codonBias > toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "AT")
+						atBiasMap[codon.Triplet] = append(atBiasMap[codon.Triplet], toCodon.Triplet)
 					case codonBias < toCodonBias:
-						tx.MustExec(`INSERT INTO codonbias(fromcodon, tocodon, gcbias) VALUES (?, ?, ?)`, codon.Triplet, toCodon.Triplet, "GC")
+						gcBiasMap[codon.Triplet] = append(gcBiasMap[codon.Triplet], toCodon.Triplet)
 					}
+					naBiasMap[codon.Triplet] = append(naBiasMap[codon.Triplet], toCodon.Triplet)
 				}
 			}
 		}
+		// aminoAcidTotal can't equal zero or else division later will panic
 		if aminoAcidTotal == 0 {
 			aminoAcidTotal = 1
 		}
-		weightTable[aminoAcid.Letter] = aminoAcidTotal
-	}
-
-	// Insert seq and history
-	pos := 0
-	for codonPosition := 0; codonPosition < len(sequence); codonPosition = codonPosition + codonLength {
-		codon := sequence[codonPosition : codonPosition+codonLength]
-		tx.MustExec(`INSERT INTO seq(pos) VALUES (?)`, pos)
-		tx.MustExec(`INSERT INTO history(pos, codon, step) VALUES (?, ?, 0)`, pos, codon)
-		pos++
+		aminoAcidWeightTable[aminoAcid.Letter] = aminoAcidTotal
 	}
 
+	// Build weight map
 	for _, aminoAcid := range codontable.AminoAcids {
```

Comment:
"Cycle through every possible amino acid."

Reply:
That's kinda what the code says already?